build-system/compile/sources.js

```diff
@@ -41,6 +41,7 @@ const COMMON_GLOBS = [
   'third_party/webcomponentsjs/ShadowCSS.js',
   'node_modules/dompurify/dist/purify.es.js',
   'node_modules/promise-pjs/promise.mjs',
+  'node_modules/rrule/dist/esm/src/*.js',
```

Comment:
I don't know a lot about `rrule`, but this line seems like it violates the norm for how external JS modules are included in the runtime because it includes all the JS files in the `src/` directory. Is there an alternative JS file in `node_modules/rrule` that contains the transitive closure of all the code that's needed in date picker?

Reply:
Is it common to only have one file here? It used to be `index.js` but that was not enough. I found that `rrulestr.js` and `parsestr.js` together seems to be OK. That resurfaces the `tslib` unknown dependency error without having to use `*.js`

Reply:
> Is it common to only have one file here?

It is, from looking at all other `node_modules` files in this list.

> It used to be `index.js` but that was not enough. I found that `rrulestr.js` and `parsestr.js` together seems to be OK. That resurfaces the `tslib` unknown dependency error without having to use `*.js`

Hmm. I'm flummoxed here. Tagging @erwinmombay, @choumx, @jridgewell, and @kristoferbaxter in case any of them know how to get around this. I'd imagine you want to isolate the functionality in `rrule` that you're using in `amp-date-picker` in as small a compilation unit as possible. Not sure what exactly that is, or if `rrule/dist/esm/` is necessarily preferable to `rrule/dist/es5/`.

Reply:
Some historical context: The original version of date-picker with `rrule` support was created by adding a [custom version](https://github.com/ampproject/amphtml/tree/2a77f2615b28acf036e1e930c7540755cdeb247d/third_party/rrule) of `rrule` to `third_party/`. Unfortunately, there was no accompanying `README.amp` file explaining how it was generated, and the file eventually had to be deleted because the version of `rrule` it was based on had security vulnerabilities.

Reply:
I'm curious why `rrule/dist` has multiple output binaries, but I don't think it's a problem in general.